src/main/java/org/usrv/http/Server.java

```diff
@@ -105,10 +107,13 @@ private void handleRequest(Socket socket) {
                         logger.debug("Create response");
                         response = new Response(200);
                         response.setHeader("Content-Type", file.getMimeType());
-                        response.setBody(body);
                         response.setHeader("Connection", "close");
                         logger.debug("Added headers");
-                        cache.put(filePath, response);
+
+                        if(!isHeadMethod) {
+                            response.setBody(body);
+                            cache.put(filePath, response);
+                        }
```

Comment:
Ideally, we wouldn't read the body in this case for performance reasons, but we can fix that in a follow up PR.